Fill.py

```diff
@@ -874,93 +851,101 @@ def failed(warning: str, force: typing.Union[bool, str]) -> None:
-            if isinstance(block['count'], int):
-                block['count'] = {'min': block['count'], 'max': block['count']}
-            if 'min' not in block['count']:
-                block['count']['min'] = 0
-            if 'max' not in block['count']:
-                block['count']['max'] = (min(len(block['items']), len(block['locations'])) if
-                                         len(block['locations']) > 0 else len(block['items']))
-            if block['count']['max'] > len(block['items']):
-                count = block['count']
-                failed(f"Plando count {count} greater than items specified", block['force'])
-                block['count'] = len(block['items'])
-            if block['count']['max'] > len(block['locations']) > 0:
-                count = block['count']
-                failed(f"Plando count {count} greater than locations specified", block['force'])
-                block['count'] = len(block['locations'])
-            block['count']['target'] = multiworld.random.randint(block['count']['min'], block['count']['max'])
-
-            if block['count']['target'] > 0:
-                plando_blocks.append(block)
+            for target_player in worlds:
+                for group in multiworld.worlds[target_player].location_name_groups:
+                    if group in locations:
+                        locations.extend(multiworld.worlds[target_player].location_name_groups[group])
+
+            new_block["locations"] = list(dict.fromkeys(locations))
+
+            new_block["count"] = block.count
+            if not new_block["count"]:
+                new_block["count"] = (min(len(new_block["items"]), len(new_block["locations"])) if
+                                      len(new_block["locations"]) > 0 else len(new_block["items"]))
+            if isinstance(new_block["count"], int):
+                new_block["count"] = {"min": new_block["count"], "max": new_block["count"]}
+            if "min" not in new_block["count"]:
+                new_block["count"]["min"] = 0
+            if "max" not in new_block["count"]:
+                new_block["count"]["max"] = (min(len(new_block["items"]), len(new_block["locations"])) if
+                                             len(new_block["locations"]) > 0 else len(new_block["items"]))
+            if new_block["count"]["max"] > len(new_block["items"]):
+                count = new_block["count"]
+                failed(f"Plando count {count} greater than items specified", block.force)
+                new_block["count"] = len(new_block["items"])
+            if new_block["count"]["max"] > len(new_block["locations"]) > 0:
+                count = new_block["count"]
+                failed(f"Plando count {count} greater than locations specified", block.force)
+                new_block["count"] = len(new_block["locations"])
+            new_block["count"]["target"] = multiworld.random.randint(new_block["count"]["min"],
```

Comment:
So this is actually already an error on `main`, but it might be worth mentioning here as well: If either of the two above `if` blocks trigger, `new_block["count"]` becomes an int, so `min` and `max` don't exist and generation fails with `TypeError: 'int' object is not subscriptable` on this line (880)

Reply:
Bump, I guess?

Reply:
Definitely thought I had changed that already.

Reply:
Oh, you totally might have, you just never replied. I can check again XD

Reply:
No it is definitely unchanged.